element/element.go

```diff
@@ -0,0 +1,36 @@
+package element
+
+import "strings"
+
+type ElementMatcher struct {
+	Elements   []string
+	Aliases    map[string][]string
+	Delimiters []string
+	ProjKey    string
+	Directory  string
+}
+
+type Matcher struct {
+	Elements   []ElementMatcher
+	CtxLines   int
+	Delimiters string
+}
+
+func (m Matcher) MatchElement(line, flagKey string) bool {
```

Comment:
This looks like it pre-supposes you want to do this work a flag key at a time.  I still question whether that is the right approach.  You could potentially take a list of keys and return what keys it matches.  That might offer a more general solution.